### PR TITLE
Mouseover Highlights for Handheld Objects

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

This adds a feature to mouseover events, it creates a clientside image to represent what you currently mouseover. This is to supplement the screentips and the tooltips at the bottom left, it is intended to make things feel more fluid in game. 

https://github.com/Bubberstation/Bubberstation/assets/77420409/51d6cceb-4398-4d30-8789-73a539c77837
VVVV- looks like this right now -VVVV
https://github.com/Bubberstation/Bubberstation/assets/77420409/debacb47-d46e-4c39-bfe3-d38569383003

## Why It's Good For The Game

Accessibility, primarily. Should let you know what exactly you are clicking on without actually clicking on said thing. Lets you see things in rubbish piles easier, for precision clicking. 

## Changelog

:cl:
add: Mouse-over highlights to world objects. 
/:cl:

### CODE REVIEW

**Comment by StrangeWeirdKitten** [comment]
Get the changelog and some documentation / pictures of what this PR does

**Comment by Majkl-J** [comment]
Please make this a pref individual of the actual highlight system I still like it when it highlights inventory items but find the whole extra layer insanely annoying

**Comment by destrucktoid** [comment]
I'm seeing a lot of complaints in mentorhelp from people saying they don't like it and i have to say i agree with them on this, i don't like this but can certainly see it being a cool feature if it had it's own separate config.

**Comment by StrangeWeirdKitten** [comment]
> I'm seeing a lot of complaints in mentorhelp from people saying they don't like it and i have to say i agree with them on this, i don't like this but can certainly see it being a cool feature (for those who DO like it) if it had it's own separate config.

There are a few things, but there's also a preference you can toggle already.

**Comment by StrangeWeirdKitten** [comment]
this needs to be fixed.

image

I've given my feedback on this and the test merge was successful in highlighting issues.

File: modular_zubbers/code/modules/mouseover/highlights.dm [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+/client/MouseExited(object, location, control, params)
+	var/initalpha = initial(mouseover_highlight_dummy.alpha)
+	var/atom/movable/current_atom = current_highlight_atom?.resolve()
+	if(current_atom != object && mouseover_highlight_dummy)
+		animate(mouseover_highlight_dummy, pixel_y = 0, time = 0.2 SECONDS, easing = ELASTIC_EASING, alpha = initalpha)
+		refresh_mouseover_highlight_timer(current_atom, object)
+
+	. = ..()
+
+
```
Comment (ReturnToZender) [inline modular_zubbers/code/modules/mouseover/highlights.dm:137]: `/datum/client_interface/var/mouseover` is missing from this file, which is causing runtimes!

**Comment by Majkl-J** [comment]
Marked stale for over a month. Reopen when work continues

**Pull request closed without merging**